upgrade/scripts/upgrade_python_package.py

```diff
@@ -204,36 +234,36 @@ def attempt_to_install_version(package_install_cmd, version, cloudsmith_url=None
     """
     attempt to install a specific version of the given package
     """
+    resp = ""
     try:
-        resp = ""
-        # constraints_file_path = install_with_no_deps(f'{package_install_cmd}=={version}')
-        # install_with_constraints(f'{package_install_cmd}=={version}', constraints_file_path)
-        install_wheel(f"{package_install_cmd}=={version}", cloudsmith_url)
-    except Exception:
+        resp = install_wheel(package_install_cmd, cloudsmith_url, version_cmd=version)
+    except Exception as e:
         logging.info(f"Could not find {package_install_cmd} {version}")
         print(f"Could not find {package_install_cmd} {version}")
-        return False
-    return "Successfully installed" in resp
+        return False, str(e)
+    return "Successfully installed" in resp, resp
 
 
-def attempt_upgrade(package_install_cmd):
+def attempt_upgrade(package_install_cmd, cloudsmith_url=None, *args):
     """
     attempt to upgrade a packgage with the given package_install_cmd.
     return True if it was upgraded.
     """
     pip_config = pip("config", "list")
     pip_args = []
-    match = development_index_re.search(pip_config)
+    match = development_index_re.search(pip_config) or "--pre" in str(args)
```

Comment:
I think that we also need to check cloudsmith url because this did not work `py -m upgrade.scripts.upgrade_python_package oll-partners --cloudsmith-url=https://dl.cloudsmith.io/xxy/openlawlibrary/development/python/index/  --format-output`

Reply:
Added